src/ruleset.hpp

```diff
@@ -22,33 +23,22 @@
 namespace ddwaf {
 
 struct ruleset {
-    void insert_rule(const std::shared_ptr<rule> &rule)
+    // OLINTNEXTLINE(bugprone-easily-swappable-parameters)
+    void insert_rules(const std::vector<std::shared_ptr<core_rule>> &base,
```

Comment:
Do you need shared ownership so that ruleset_builder can reuse the rules in updates that don't replace the rules? Did you ensure that rules that are mutated in an update (through toggle/set_actions/set_verdict) are not incorrectly shared between rulesets?

Reply:
> Do you need shared ownership so that ruleset_builder can reuse the rules in updates that don't replace the rules? 
Yes, and between live handles / contexts during or after update.

> Did you ensure that rules that are mutated in an update (through toggle/set_actions/set_verdict) are not incorrectly shared between rulesets?
Currently rules are not mutated, they are regenerated when overrides are applied.